__init__.py

```diff
@@ -148,6 +152,38 @@ def all(self, args):
     validate_args().all(args)
 
     ### Run backend to get data
-    extracted_data = main_backend(args)
-
-    main_frontend(extracted_data, args)
\ No newline at end of file
+    df, df2, extracted_data = main_backend(args)
+
+    main_frontend(extracted_data, args)
+
+    # print(df.columns)
+    # print(df2.columns)
+
+    df2['SubmitDate'] = df2['SubmitDatetimeX'].dt.date
+
+    df2['TotalCPU (hours)'] = df2.apply(calc_CPU, axis=1)
+    df2['CPUtime (hours)'] = df2.apply(calc_CPUtime, axis=1)
+    df2['Efficientcy (TotalCPU / CPUtime hours)'] = df2.apply(calc_efficiency, axis=1)
+    df2['Cost (Pounds)'] = df2.apply(calc_cost, axis=1)
+
+    df2.to_csv(f"{args.userCWD}/{args.user}_all_data.csv", index=False)
+
+    result_submitdate = df2.groupby('SubmitDate').agg({
+    'carbonFootprint': 'sum',
```

Comment:
to avoid duplication, maybe create a dictionary :
```
groupby_dict = {
    'carbonFootprint': 'sum',
    'carbonFootprint_failedJobs': 'sum',
    'cost': 'sum',
    'cost_memoryNeededOnly': 'sum',
    'cost_failedJobs': 'sum',
    'Cost (Pounds)': 'sum',
    'UserX': 'first'}
```

and then pass it to :
`result_jobname = df2.groupby('JobName').agg(groupby_dict )....`
`result_submitdate = df2.groupby('SubmitDate').agg(groupby_dict)...`